src/main/js/components/subcomponents/DiningSession.js

```diff
@@ -23,4 +35,109 @@ export class CustomerDiningSessionSelect extends React.Component {
+                diningSessionAttributes={this.props.diningSessionAttributes}
+                history = {this.props.history}
+                onCreate={this.onCreate}
+                onUpdate={this.onUpdate}
+                onDelete={this.onDelete}
+                onNavigate={this.onNavigate}
+            />)
+
+        //TODO review how to get number of diningSessions. Test whether this.props.diningSessions.size returns a number or undefined
+        return  (
+            <div>
+                <title>All Orders</title>
+                {sessions}
+                
+            </div>
+        )
+    }
+
+    handleClose(){
+        //load orders
+        // const orders;
+    }
+
+}
+
+export class StaffDiningSession extends React.Component{
+
+    constructor(props){
+        super(props);
+        this.handleListOrders = this.handleListOrders.bind(this);
+    }
+
+    requestOrders(){
+        fetch(this.props.diningSession.entity._links.orders.href, {method: 'GET', headers: {'Content-Type': 'application/json'}})
+            .then(
+                response => {
+                    if (!response.ok) {
+                        console.log('Looks like there was a problem. Status Code: ' +
+                            response.status);
+                        return;
+                    }
+
+                    // Examine the text in the response
+                    response.json().then((data) => {
+                        console.log(data._embedded.orders);
+                        this.setState(
+                            {
+                                orders : data._embedded.orders
+                            }
+                        );
+                    });
+                }
+            )
+            .catch(function(err) {
+                console.log('Fetch Error :-S', err);
+            });
+   }
+    render(){
+        // console.log("Individual session:");
+        // console.log(this.props.diningSession);
+        return (
+            <div className="gridViewItem">               
+                    <img className="itemImage" draggable="false" src="./img/3.jpg" />               
+                    <div className="overlay">
+                        <div className="text">Table {this.props.diningSession.entity.tableNumber}</div>
+                        <div className="text">{this.props.diningSession.entity.price}</div>
+                        <div style={{display: 'flex', justifyContent: 'center'}}>
+                            <button className="view-detail-button" title="View details">
+                                <i className="view-order" style={{fontSize: '20px'}} onClick={this.handleListOrders}>View</i>
+                            </button>
+                        </div>
+                        
+                    </div> 
+            </div>
+        )
+    }
+
+    handleListOrders(){
+        this.props.history.push
+        (
+            {
+            pathname: ('/orders/'),
```

Comment:
Might be a bit nit-picky but is there a reason/need for the terminal `/` in your pathname? all other pathnames end without a `/`.